### DISCUSSION TITLE
Different plotting options

### DISCUSSION THREAD

**Discussion by Sky-creates** [qcha41/usit#37] [Ideas]
I want to monitor a variable x, which is a 2D numpy array with size (2,1000). I hope to plot it as two 1D plots consists of 1000 points, something like plot(time,x[0]); plot(time,x[1]), instead of monitoring as a 2D image. Is there a quick way I can do this? It would be best if we can extend to 2D arrays with similar sizes :)
Thanks! I also want to say that autolab is a great project.

**Comment by Python-simulation** [comment]
Hi! Thank you for your input! Currently the monitor cannot do it, but I would be a nice feature to add.

First of all, in autolab the convention we choose for arrays is (1000, 2) instead of (2, 1000) which, in your case, results in a 2D image instead of a plot line of the second axis with respect to the first one, plot(x[:, 0], x[:, 1]).

Another solution that could be useful with ND arrays is to use the plotter instead, which gives more control over the different axes to plot.
But in your case, it would not give you the wanted results plot(time, x[:, 0]) but only plot(x[:, i], x[:, j]). (You could convert your ndarray to a pd.DataFrame with three column, first for id (1,2,3,4,...) second and third for your two 1000 points).

What you can do right now is to create two different variables in your driver returning only the 1000 points array to be monitored separately.
These variables could just be a reading of an internal value in the driver set by your 2D array variable write function avoiding sending twice the acquisition command to the instrument. In this case, you need to monitor the 2D array for the acquisition and monitor the two variables for display.

I will add the axis selection feature in the todo list.

**Reply by FateDecider** [reply]
Hi, regarding your suggestion to monitor two line plots, I suppose you mean to do this on 'Scanner' rather than 'Monitor'. 

Because on the control panel, it is not clear to me how to open the Monitor window for these two different variables without sending twice the acquisition command to the instrument.

**Reply by Python-simulation** [reply]
I was not referring to the scanner but to the monitor only. The idea, which is not ideal, would be to monitor something blank that do acquisition (first monitor) and then have the two monitors that get the stored data for the two lines (second and third monitors).
This solution is not my favorite. I would instead suggest to use the Plotter to get the 2D array and select the proper axis.

**Reply by FateDecider** [reply]
I see, yeah I figured you have to do such a work around to access the internal value set in the driver. 

For myself, I am thinking of monitoring a one-dimensional array (like a waveform from an oscilloscope) across different channels on the instrument. But I think at present, monitoring multiple channels simultaneously and acquiring smooth updates for a big 1D array on the monitor window can prove challenging.
